mlrun/runtimes/sparkjob/abstract.py

```diff
@@ -363,6 +363,15 @@ def _get_spark_version(self):
     def _get_igz_deps(self):
         raise NotImplementedError()
 
+    def _pre_run(self, runobj: RunObject, execution: MLClientCtx):
```

Comment:
better do validations on server side

Reply:
In general that's true, but basic validations should be performed as early as possible - it serves to give a better UX, failing the command immediately rather than reflecting an 404 error which includes the actual reason in it, after the server rejects the request (not to mention the extra processing and function saving that is happening before that, and should be avoided in this case)